### ISSUE TITLE
[v2] incorrect comment loc info

### ISSUE THREAD

**Issue by JounQin** [mdx-js/eslint-mdx#380]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.0.next.1

### Steps to reproduce

````mdx
{/* First */}

<Story>
  {{
    template: /* HTML */ ``,
  }}
</Story>
````

print with `console.log(JSON.stringify(node.data?.estree.comments || [], null, 2))`

### Expected behavior

The second comment should start with `line: 5, column: 14` and end with `line: 5, column: 24`

### Actual behavior

```json
[
  {
    "type": "Block",
    "value": " First ",
    "start": 1,
    "end": 12,
    "loc": {
      "start": {
        "line": 1,
        "column": 1
      },
      "end": {
        "line": 1,
        "column": 12
      }
    },
    "range": [
      1,
      12
    ]
  }
]
[
  {
    "type": "Block",
    "value": " HTML ",
    "start": 42,
    "end": 52,
    "loc": {
      "start": {
        "line": 4,
        "column": 19
      },
      "end": {
        "line": 4,
        "column": 29
      }
    },
    "range": [
      42,
      52
    ]
  }
]
```

### Runtime

Node v16

### Package manager

yarn v1

### OS

macOS

**Comment by JounQin** [comment]
@wooorm Do you have any idea? The `node` is `unist Node` from `remark-*` at https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-mdx/src/parser.ts#L89

The worst thing is the wrong comment loc may block ESLint process(I didn't find out why yet).

**Comment by wooorm** [comment]
- any reason you’re not on a stable v2?
- parsing of each “chunk” of JS-in-markdown happens from 1:1, and then a starting position as added onto it, I guess this is happening for nodes, but not for comments, here: https://github.com/micromark/micromark-extension-mdx-expression/blob/3d988bc4d5db5fbafb4cf5f719e66c25994bfb3f/packages/micromark-util-events-to-acorn/dev/index.js#L148-L187

**Comment by JounQin** [comment]
> any reason you’re not on a stable v2?

`2.0.0.next.1` is `eslint-mdx`

> parsing of each “chunk” of JS-in-markdown happens from 1:1, and then a starting position as added onto it, I guess this is happening for nodes, but not for comments

So this is a known issue? Are you going to fix it?

**Comment by wooorm** [comment]
I can imagine it being a problem.
And if it’s an issue for you, then it’s now a known problem :)

Do you want to investigate? It might be useful for the future to get your hands at micromark-extension-mdx*?
But: I don’t mind doing it either, it’s probably faster if I do it!

**Comment by JounQin** [comment]
OK, I will give it a try today.

**Comment by JounQin** [comment]
@wooorm See https://github.com/micromark/micromark-extension-mdx-expression/pull/2

**Comment by wooorm** [comment]
Perhaps you can make a small reproduction of this problem?

**Comment by JounQin** [comment]
@wooorm See https://github.com/JounQin/test/tree/mdx-loc

branch `mdx-loc`, simply run `npm test` to see comments loc info.

**Comment by wooorm** [comment]
The first comment is correct, I believe?
Do you have a problem with the second comment?

**Comment by wooorm** [comment]
Something fishy going on, because the earlier value affects the second comment:

```
{/* First */}

<Story>
  {{
    template: /* HTML */ ``,
  }}
</Story>
```

```
{
  type: 'Block',
  value: ' HTML ',
  start: 42,
  end: 52,
  loc: {
    start: { line: 4, column: 19 },
    end: { line: 4, column: 29 }
  },
  range: [ 42, 52 ]
}
```

---

```
{}

<Story>
  {{
    template: /* HTML */ ``,
  }}
</Story>
```

```
{
  type: 'Block',
  value: ' HTML ',
  start: 31,
  end: 41,
  loc: {
    start: { line: 4, column: 19 },
    end: { line: 5, column: 24 }
  },
  range: [ 31, 41 ]
}
```

**Comment by JounQin** [comment]
> The first comment is correct, right?

Should the `column` of `unified` node be `1`-indexed? (`acorn`'s is `0`-indexed).

> Btw, I don’t think the code makes sense? Why is there an object as the child of Story

This is how `storybook` works.

**Comment by wooorm** [comment]
I’m not sure how I came up with the previous logic. I made a little algorithm locally that works and is much simpler. I’ll have to make the code pretty and add test it all, but should have it ready soon!

**Comment by wooorm** [comment]
Solved in https://github.com/micromark/micromark-extension-mdx-expression/releases/tag/micromark-util-events-to-acorn%401.0.5!

**Comment by JounQin** [comment]
@wooorm

````mdx
import Basic from './basic'

export const meta={
  title: 'Blog Post',
}

# Blog Post

Lorem ipsum dolor sit amet, consectetur adipiscing **elit**. Ut ac lobortis <b>velit</b>.

{/* This is a comment */}

```css
@media (min-width: 400px) {
  border-color: #000;
}
```

<Basic>{/* This is a comment */}</Basic>

## Subtitle

Lorem ipsum dolor sit _amet_, consectetur adipiscing elit. Ut ac lobortis velit.
````

```log
TypeError: Cannot read properties of undefined (reading 'column')
          at parseOffsetToUnistPoint (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark-util-events-to-acorn/index.js:212:32)
          at file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark-util-events-to-acorn/index.js:168:26
          at visit (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/estree-util-visit/index.js:85:37)
          at visit (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/estree-util-visit/index.js:68:30)
          at eventsToAcorn (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark-util-events-to-acorn/index.js:148:5)
          at atEnd (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark-extension-mdxjs-esm/lib/syntax.js:144:22)
          at go (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark/lib/create-tokenizer.js:216:13)
          at main (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark/lib/create-tokenizer.js:202:9)
          at Object.write (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark/lib/create-tokenizer.js:132:5)
          at writeToChild (file:///Users/JounQin/Workspaces/GitHub/eslint-mdx/node_modules/micromark/lib/initialize/document.js:247:15)
```

**Comment by wooorm** [comment]
done
